lib/src/components/NxForm/__tests__/NxForm.test.tsx

```diff
@@ -22,268 +19,298 @@ describe('NxForm', function() {
+      expect(el).toHaveClass('nx-form');
     });
 
     it('sets extra attrs on the form', function() {
-      const component = getShallow({ id: 'foo', lang: 'en_US' });
+      const el = renderEl({ id: 'foo', lang: 'en_US' });
 
-      expect(component).toHaveProp('id', 'foo');
-      expect(component).toHaveProp('lang', 'en_US');
+      expect(el).toHaveAttribute('id', 'foo');
+      expect(el).toHaveAttribute('lang', 'en_US');
     });
 
     it('puts the ref on the form', function() {
       const ref = React.createRef<HTMLFormElement>(),
+          el = renderEl({ ref });
 
-          // note: the fragment is necessary to get around an enzyme issue:
-          // https://github.com/enzymejs/enzyme/issues/1852#issuecomment-433145879
-          component = mount(<><NxForm { ...minimalProps } ref={ref} /></>),
-          domNode = component.find('form').getDOMNode();
-
-      expect(ref.current).toBe(domNode);
+      expect(ref.current).toBe(el);
     });
 
     describe('on form submit', function() {
-      let component: ShallowWrapper,
-          onSubmit: jest.Mock;
+      it('does not execute a native form submission', async function() {
+        const form = renderEl()!;
+
+        const defaultNotPrevented = fireEvent.submit(form);
 
-      beforeEach(function() {
-        onSubmit = jest.fn();
-        component = getShallow({ onSubmit });
+        expect(defaultNotPrevented).toBe(false);
       });
 
-      it('calls preventDefault on the event', function() {
-        const event = { preventDefault: jest.fn() };
+      it('calls the onSubmit prop', async function() {
+        const onSubmit = jest.fn(),
+            form = renderEl({ onSubmit })!;
 
-        component.simulate('submit', event);
-        expect(event.preventDefault).toHaveBeenCalled();
-      });
+        expect(onSubmit).not.toHaveBeenCalled();
 
-      it('calls the onSubmit prop', function() {
-        component.simulate('submit', { preventDefault: jest.fn() });
+        fireEvent.submit(form);
 
         expect(onSubmit).toHaveBeenCalled();
       });
     });
 
     it('renders children', function() {
-      expect(getShallow({ children: <div id="foo" /> })).toContainMatchingElement('div#foo');
+      expect(quickRender({ children: <div data-testid="foo" /> }).getByTestId('foo')).toBeInTheDocument();
     });
 
     it('renders children passed as a function', function() {
-      expect(getShallow({ children: () => <div id="foo" /> })).toContainMatchingElement('div#foo');
+      expect(quickRender({ children: () => <div data-testid="foo" /> }).getByTestId('foo')).toBeInTheDocument();
     });
 
     it('provides the showValidationErrors value to the children via FormAriaContext', function() {
       function Fixture() {
         const { showValidationErrors } = useContext(FormAriaContext);
 
-        return <span className="fixture">{showValidationErrors.toString()}</span>;
+        return <span data-testid="foo">{showValidationErrors.toString()}</span>;
```

Comment:
Interesting work-around to follow RTL's 'toHaveTextContent' matcher. For future reference, is this the preferred way (instead of asserting .toBeTrue() ) when moving forward with RTL?

Reply:
> Interesting work-around to follow RTL's 'toHaveTextContent' matcher. For future reference, is this the preferred way (instead of asserting .toBeTrue() ) when moving forward with RTL?

As far as I can tell `toBeTrue` isn't an option here.  If it was it's what I'd prefer.  But what I'm doing here is creating a child component that uses the context, and who's render output I can use to inspect the context values.  But the render is of course DOM and strings, so I have to assert the value as a string

Reply:
> As far as I can tell `toBeTrue` isn't an option here. If it was it's what I'd prefer. But what I'm doing here is creating a child component that uses the context, and who's render output I can use to inspect the context values. But the render is of course DOM and strings, so I have to assert the value as a string

Asserting the value as a string was really clever. Thanks for the explanation